ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -502,6 +510,7 @@ void initiateChunkDeletesIfAny(final String blobIdStr, final String serviceId,
           deleteRequests.add(new BackgroundDeleteRequest(storeKey, serviceId, quotaChargeCallback));
         }
         initiateBackgroundDeletes(deleteRequests);
+        blobMetadataCache.deleteObject(blobIdStr);
```

Comment:
it's good to remove it from cache in delete, but I think we should remove it from cache in ttl update as well, because ttl update would change the ttl for metadata. So the blob is outdate after ttl update.

Reply:
Yeah we discussed this and Tugrul suggested letting the cache entry remain and let the frontend talk to backend to validate the cache entry during the next RANGE request.

Reply:
Actually, this reminds me that if the blob doesn't exist in the backend, we should invalidate its cache entry in the frontend. I need to add that code.